cocos/audio/src/AudioDecoderEXT.mm

```diff
@@ -158,4 +158,53 @@ of this software and associated documentation files (the "Software"), to deal
         } while(false);
         return ret;
     }
+
+    void AudioDecoderEXT::closeInternal()
+    {
+        if (_extRef != nullptr)
+        {
+            ExtAudioFileDispose(_extRef);
+            AudioFileClose(_audioFileId);
+            _extRef = nullptr;
+            _audioFileId = nullptr;
+            _fileStream = nullptr;
+        }
+    }
+
+    OSStatus AudioDecoderEXT::readCallback(void *inClientData, SInt64 inPosition, UInt32 requestCount, void *buffer, UInt32 *actualCount)
+    {
+        if (!inClientData)
+        {
+            return kAudioFileNotOpenError;
+        }
+        
+        auto* fileStream = (cocos2d::FileStream*)inClientData;
+                
+        const auto seekResult = fileStream->seek(inPosition, SEEK_SET);
+        if (seekResult < 0)
+        {
+            return kAudioFilePositionError;
+        }
+        
+        const auto count = fileStream->read(buffer, requestCount);
+        
+        if (count < 0)
+        {
+            return kAudioFileEndOfFileError;
+        }
+        
+        *actualCount = count;
+        
+        return noErr;
+    }
+
+    SInt64 AudioDecoderEXT::getSizeCallback(void *inClientData)
+    {
+        auto* fileStream = (cocos2d::FileStream*)inClientData;
+        //auto currentPos = fileStream->tell();
+        fileStream->seek(0, SEEK_END);
+        auto fileSize = (SInt64)fileStream->tell();
+        //fileStream->seek(currentPos, SEEK_SET); //go back to where we were
```

Comment:
Why this line is commented?

Reply:
Oh sorry I was testing if the file position needed to be set back to the original location before that getSize is called, but it seems that it's only ever called once at the start, so no need to reset the position again. I'll delete the commented code.